extensions/amp-fx-collection/0.1/providers/amp-fx-presets.js

```diff
@@ -65,6 +67,194 @@ export const Presets = {
+    isFxTypeSupported(win) {
+      return isExperimentOn(win, 'amp-fx-fly-in');
+    },
+    userAsserts(element) {
+      const marginStart = parseFloat(element.getAttribute('data-margin-start'));
+      if (!marginStart) {
+        return;
+      }
+      user().assert(marginStart >= 0 && marginStart <= 100,
+          'data-margin-start must be a percentage value ' +
+          'and be between 0% and 100% for: %s', element);
+    },
+    update(entry) {
+      const fxElement = this;
+      dev().assert(fxElement.adjustedViewportHeight);
+      const top = entry.positionRect ? entry.positionRect.top : null;
+      // Outside viewport
+      if (!top || top - (fxElement.adjustedViewportHeight *
+        fxElement.getFlyInDistance() / 100) >
+          (1 - fxElement.getMarginStart()) *
+            fxElement.adjustedViewportHeight) {
+        return;
+      }
+
+      if (fxElement.isMutateScheduled()) {
+        return;
+      }
+
+      // If above the threshold of trigger-position
+      fxElement.setIsMutateScheduled(true);
+      fxElement.resources_.mutateElement(fxElement.getElement(), function() {
+        fxElement.setIsMutateScheduled(false);
+        // Translate the element offset pixels.
+        setStyles(fxElement.getElement(), {
+          'transition-duration': fxElement.getDuration(),
+          'transition-timing-function': fxElement.getEasing(),
+          'transform': `translateY(-${fxElement.getFlyInDistance()}vh)`,
+        });
+      });
+    },
+  },
+  'fly-in-left': {
+    isFxTypeSupported(win) {
+      return isExperimentOn(win, 'amp-fx-fly-in');
+    },
+    userAsserts(element) {
+      const marginStart = parseFloat(element.getAttribute('data-margin-start'));
+      if (!marginStart) {
+        return;
+      }
+      user().assert(marginStart >= 0 && marginStart <= 100,
+          'data-margin-start must be a percentage value ' +
+          'and be between 0% and 100% for: %s', element);
+    },
+    update(entry) {
+      const fxElement = this;
+      dev().assert(fxElement.adjustedViewportHeight);
+      const top = entry.positionRect ? entry.positionRect.top : null;
+      // Outside viewport
+      if (!top || top > (1 - fxElement.getMarginStart()) *
+        fxElement.adjustedViewportHeight) {
+        return;
+      }
+
+      if (fxElement.isMutateScheduled()) {
+        return;
+      }
+
+      // only do this on the first element
+      if (!fxElement.initialTrigger) {
+        const style = computedStyle(fxElement.getAmpDoc().win,
+            fxElement.getElement());
+        setStyles(fxElement.getElement(), {
+          'left':
+            'calc(' + style.left + ' - ' + fxElement.getFlyInDistance() + 'vw)',
+          'visibility': 'visible',
+        });
+        fxElement.initialTrigger = true;
+      }
+
+      // If above the threshold of trigger-position
+      fxElement.setIsMutateScheduled(true);
+      fxElement.resources_.mutateElement(fxElement.getElement(), function() {
```

Comment:
Can't access private prop :)

Reply:
Thanks!

Reply:
I don't see the change?

Reply:
The latest commit has the changes - https://github.com/ampproject/amphtml/pull/14782/files#diff-2002eb64c8e37a740e78da7561f5ce2eR171